File: mssp/examples/002-link-checker/src/TMS/reporters/text.py
"""Renders a report for a terminal.

Reads only SMS types. It never asks which checker produced a verdict in order to
format it differently, which is what keeps adding a checker from being a change
here as well.
"""
from __future__ import annotations

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2]))

from SMS.model import Report  # noqa: E402

name = "reporters/text"


def render(report: Report) -> str:
    lines = []
    for v in report.verdicts:
        mark = "ok  " if v.ok else "FAIL"
        lines.append(f"  {mark} {v.link.page} -> {v.link.target}  ({v.detail})")
    for link in report.unchecked:
        lines.append(f"  ????  {link.page} -> {link.target}  (no verdict)")

    failed = sum(1 for v in report.verdicts if not v.ok)
    lines.append("")
    lines.append(
        f"  {len(report.verdicts)} checked, {failed} failed, "
        f"{len(report.unchecked)} left without a verdict"
    )
    lines.append(f"  loop closed: {report.closed}")
    return "\n".join(lines)

File: mssp/examples/005-before-after/src/TMS/formats/text.js
// Renders a summary for a terminal. Reads SMS shapes only.
export const name = "formats/text";

export function render(summary, dropped) {
  const lines = summary.map(
    (s) => `  ${s.sensor.padEnd(8)} n=${s.n}  mean=${s.mean}  range=${s.min}..${s.max}`,
  );
  lines.push(`  ${dropped.length} row(s) dropped`);
  return lines.join("\n");
}
